core/server/worker/src/main/java/alluxio/worker/AlluxioWorkerRestServiceHandler.java

```diff
@@ -464,7 +464,8 @@ public Response getWebUILogs(@DefaultValue("") @QueryParam("path") String reques
       @QueryParam("end") String requestEnd,
       @DefaultValue("20") @QueryParam("limit") String requestLimit) {
     return RestUtils.call(() -> {
-      FilenameFilter filenameFilter = (dir, name) -> name.toLowerCase().endsWith(".log");
+      FilenameFilter filenameFilter = (dir, name) -> name.toLowerCase()
+          .matches(".*\\.log[\\d+]|.*.out|.*.txt|.*.json");
```

Comment:
string.matches compiles an internal Pattern state machine every time. Could you create a reused `Pattern` obj in this class in `static`? Also could you add a UT on that?